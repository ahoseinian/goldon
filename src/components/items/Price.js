import React from 'react'
import {Small} from '../common/typography/Small'

const Price = ({value}) => (
  <div>
    {value.toLocaleString()}
    <Small>تومان</Small>
  </div>
)

export default Price
