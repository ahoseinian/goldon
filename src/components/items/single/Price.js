import React from 'react';
import styled from 'styled-components'
import {color} from '../../theme'

const Container = styled.div`
  background: ${color.light};
  padding: 1.5rem;
`

const Currency = styled.small`
  position: relative;
  top: 0.4rem;
  margin-right: 1rem;
`

const Price = ({amount}) => (
  <Container className="d-flex justify-content-center h4">
    <div>{amount.toLocaleString()}</div>
    <Currency>تومان</Currency>
  </Container>
);

Price.propTypes = {
  amount: React.PropTypes.number.isRequired
};

export default Price;
