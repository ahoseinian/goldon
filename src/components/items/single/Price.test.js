import React from 'react'
import {shallow} from 'enzyme'
import Price from './Price'
import toJson from 'enzyme-to-json'

describe('Price', () => {
  it('renders ', () => {
    const wrapper = shallow(<Price amount={2000}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
