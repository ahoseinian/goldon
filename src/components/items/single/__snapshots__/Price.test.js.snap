exports[`Price renders  1`] = `
<styled.div
  className="d-flex justify-content-center h4 m-0 h-100">
  <div
    className="price">
    2,000
  </div>
  <styled.small>
    تومان
  </styled.small>
</styled.div>
`;
